src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +157,21 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        $salt = '';
+
+        if ($this->getConfig('salt')) {
+            if (is_string($this->getConfig('salt'))) {
+                $salt = $this->getConfig('salt');
+            } else {
+                $salt = Security::getSalt();
+            }
+
+            if (!is_string($salt) || strlen($salt) === 0) {
+                throw new InvalidArgumentException('Salt must be a non-empty string.');
+            }
+        }
+
+        return $identity[$usernameField] . $identity[$passwordField] . $salt;
```

Comment:
```suggestion
        $value = $identity[$usernameField] . $identity[$passwordField];
        $hmac = hash_hmac('sha1', $value, $salt);

        return $value . $hmac;
```

I know the method says `Plain`, suggesting that the return value must not be used as-is, but I'd still feel better if the salt was kept as "internal" as possible and only used to "sign" data.

Reply:
I don't understand why you're hashing here, the value that comes out of this method is hashed with the password hasher before being used in a cookie. Wouldn't that be enough?

Reply:
Good point about the risk of having the bare salt included.

Reply:
Thanks for the suggestion, changed it.